python/pylibcudf/pylibcudf/strings/char_types.pyx

```diff
@@ -1,4 +1,92 @@
 # Copyright (c) 2024, NVIDIA CORPORATION.
 
+from libcpp.memory cimport unique_ptr
+from libcpp.utility cimport move
+from pylibcudf.column cimport Column
+from pylibcudf.libcudf.column.column cimport column
+from pylibcudf.libcudf.scalar.scalar cimport string_scalar
+from pylibcudf.libcudf.strings cimport char_types as cpp_char_types
+from pylibcudf.scalar cimport Scalar
+
+from cython.operator import dereference
 from pylibcudf.libcudf.strings.char_types import \
     string_character_types as StringCharacterTypes  # no-cython-lint
+
+
+cpdef Column all_characters_of_type(
+    Column source_strings,
+    string_character_types types,
+    string_character_types verify_types
+):
+    """
+    Filter specific character types from a column of strings.
+
+    Parameters
+    ----------
+    source_strings : Column
+        Strings instance for this operation
+    types : StringCharacterTypes
+        The character types to check in each string
+    verify_types : StringCharacterTypes
+        Only verify against these character types.
+
+    Returns
+    -------
+    Column
+        New column of boolean results for each string
+    """
+    cdef unique_ptr[column] c_result
+
+    with nogil:
+        c_result = move(
+            cpp_char_types.all_characters_of_type(
+                source_strings.view(),
+                types,
+                verify_types,
+            )
+        )
+
+    return Column.from_libcudf(move(c_result))
+
+cpdef Column filter_characters_of_type(
+    Column source_strings,
+    string_character_types types_to_remove,
+    Scalar replacement,
+    string_character_types types_to_keep
+):
+    """
+    Filter specific character types from a column of strings.
+
+    Parameters
+    ----------
+    source_strings : Column
+        Strings instance for this operation
+    types_to_remove : StringCharacterTypes
+        The character types to check in each string.
+    replacement : Scalar
+        The replacement character to use when removing characters
+    types_to_keep : StringCharacterTypes
+        Default `ALL_TYPES` means all characters of `types_to_remove`
+        will be filtered.
+
+    Returns
+    -------
+    Column
+        New column of boolean results for each string
```

Comment:
This is wrong, it returns a new strings column with the requested characters filtered out and replaced with replacement.

I think you've copied this from the C++ docstring, which, it turns out, is also wrong, could you update that one too please?